src/components/common/DiscordGuildSetup/DiscordGuildSetup.tsx

```diff
@@ -93,7 +94,10 @@ const DiscordGuildSetup = ({
               onSelect={
                 selectedServer
                   ? undefined
-                  : (newServerId) => setValue(fieldName, newServerId)
+                  : (newServerId) => {
```

Comment:
We can revert this change and use the `useGateables` (for Discord) or `useIsTGBotIn` (for Telegram) hook inside the `BasicInfo` component in order to set the guild image from Discord server image / Telegram group image

\+ you can remove the `useSetRoleImageAndNameFromPlatformData` hook from the `TelegramGroup` & `ServerSetupCard` components, we don't need to set the image/name from there anymore